crates/datasources/src/excel/table.rs

```diff
@@ -0,0 +1,64 @@
+use std::borrow::Cow;
+use std::sync::Arc;
+
+use calamine::{open_workbook, Reader, Xlsx};
+use datafusion::datasource::streaming::StreamingTable;
+use datafusion::datasource::TableProvider;
+use datafusion::physical_plan::streaming::PartitionStream;
+
+use super::xlsx_sheet_value_to_record_batch;
+use crate::common::url::DatasourceUrl;
+use crate::excel;
+use crate::excel::errors::ExcelError;
+use crate::excel::stream::ExcelSheetPartitionStream;
+use crate::object_store::generic::GenericStoreAccess;
+
+pub async fn excel_streaming_table(
```

Comment:
so I really like the streaming table model that DF provides, _but_ they are pretty simple, and offer the greatest amount of benefit when there are multiple files (streams) or very large files that can't be in memory for whatever reason. 

In the case of these excel files, there's only ever going to be one file, and I think all or most of it has to be in memory to execute the query, so we might be able to get better results by:
writing our own query plan manually, take advantage of pushdown operations, and support more optimization in the future.

Reply:
I kept going back and forth on the mapping here and couldn't decide, even if there were multiple sheets supported it'd all be read into memory anyway by calamite, so yes agree that extending at the query plan makes sense.

Reply:
well or just providing an implementation of `TableProvider` and `ExecutionPlan` rather than using the streaming version. I think deep down the code doesn't change just gets moved around a bit,